content/blog/creating-dscc-block-storage-resources-using-dscc-ansible-playbooks.md

```diff
@@ -0,0 +1,193 @@
+    - operating_system: RHE Linux
+    - workload: ORACLE_LOG
+    - volume_set_name: ansibleVolumeSet
+    - volume_name: ansibleVolume
+```
+
+Contents of greenlake_config.json:
+
+```json
+{
+  "host": "https://us1.data.cloud.hpe.com",
+  "client_id": "009e17ef-c356-4066-8b66-4ee10e1128b1",
+  "client_secret": "53a9dd94d61311ec9b7e3ad2f9067b26"
+}
+```
+
+**Creation of host**
+
+A host represents a physical server and a host group represents a collection of physical servers. In the context of DSCC API, a host is a definition for a group of initiators that belong to a single server and a host group is a group of initiators across servers. 
+
+To create a host, initiator information is needed and this can be obtained by a GET call to get the initiator resources. One of the initiators can be used in the creation of the host.
+This means that the creation of a host requires two calls;, one to get the initiator and another  to create a host. Along with this, you must mention the name of the host, operating system, and a “user_created” flag must be set to true. Variable names will be mentioned between two curly braces and double quotes as shown in the below code snippet.
+
+```yaml
+    - name: Get GreenLake Host Initiators
+      greenlake_host_initiator_facts:
+        config: "{{ config }}"
+
+    - debug: var=host_initiators
+
+    - set_fact:
+          initiators='{{ initiators + [item.id] }}'
+      loop: "{{ host_initiators }}"
+      when: item.protocol == 'FC' and item.hosts|length <= 0
+
+    - debug: var=initiators
+
+    - name: Create GreenLake DSCC Host
+      greenlake_host:
+        config: "{{ config }}"
+        state: present
+        data:
+          initiator_ids:
+            - "{{initiators.0}}"
+          name: "{{host_name}}"
+          operating_system: "{{operating_system}}"
+          user_created: True
+
+    - debug: var=hosts
+```
+
+**Creation of host group**
+
+A host group can be created using the host created in the previous step. The tasks of this Ansible playbook will be executed sequentially, so the creation of the host group task executes only after the host creation task is successfully completed.
+
+Provide the name of the host group, host Id, and make sure that the “user_created” flag is set to true.
+
+```yaml
+    - name: Create GreenLake DSCC Host Group
+      greenlake_host_group:
+        config: "{{ config }}"
+        state: present
+        data:
+          name: "{{host_group_name}}"
+          hostIds:
+            - "{{hosts.0.id}}"
+          user_created: True
+
+    - debug: var=host_groups
+```
+
+**Creation of Volume Set**
+
+The device type represents whether it’s a Primera or Nimble volume. If it is 1, then it is a Primera volume and if it is 2, then it is a Nimble volume. Mandatory parameters are app_set_name, app_set_importance, app_set_type (which is the kind of workload that is required, like Oracle Database). Optional parameters are commented on in the below snippet.
```

Comment:
Make sure to escape the special character "_" for the variable names.

For the device type, shouldn't you add anything about Alletra 9K/Alletra 6K in addition to Primera and Nimble?

I might be wrong here, but I believe that Device type 1 relates to HPE Primera/HPE Alletra 9K storage device, and Device type 2 relates to HPE Nimble/HPE Alletra 6K storage devices.
You may want to double check with your team.

Reply:
Changed it. Thanks.,